src/compiler/scala/tools/nsc/CompileSocket.scala

```diff
@@ -138,19 +127,19 @@ class CompileSocket extends CompileOutputCommon {
     }
     info("[Port number: " + port + "]")
     if (port < 0)
-      fatal("Could not connect to compilation daemon after " + attempts + " attempts.")
+      fatal(s"Could not connect to compilation daemon after $attempts attempts. To run without it, use `-nocompdaemon` or `-nc`.")
     port
   }
 
   /** Set the port number to which a scala compile server is connected */
-  def setPort(port: Int) {
+  def setPort(port: Int): Unit = {
     val file    = portFile(port)
-    val secret  = new SecureRandom().nextInt.toString
+    val secretBytes = new Array[Byte](16)
+    new SecureRandom().nextBytes(secretBytes)
+    val secretDigits = new BigInteger(secretBytes).toString().getBytes("UTF-8")
```

Comment:
these 3 lines of code are quite mystifying. some comments explaining the motivation would be good